src/app/clusters/door-lock-server/door-lock-server.cpp

```diff
@@ -628,7 +629,7 @@ void DoorLockServer::setCredentialCommandHandler(
 
         status = createCredential(commandPath.mEndpointId, fabricIdx, sourceNodeId, credentialIndex, credentialType,
                                   existingCredential, credentialData, userIndex, userStatus, userType, createdUserIndex);
-        sendSetCredentialResponse(commandObj, commandPath, status, createdUserIndex, nextAvailableCredentialSlot);
+        sendSetCredentialResponse(commandObj, commandPath, status, userIndex.Value(), nextAvailableCredentialSlot);
```

Comment:
And _here_ is the one place we should return a nonzero user-index, and it should be the _output from the add_, not the input.

Reply:
4.41.2. UserIndex
The user index that was created with the new credential. This SHALL be null if OperationType was Modify; otherwise, this SHALL NOT be null and SHALL provide the UserIndex created.

I see what you are saying. From reading the spec, it seems that on line 632, we should send the actual userIndex (the one used as an array index) because of 4.41.2

But I guess 4.41.2 indicates that this should be null for the modify response. So might need to change line 644?

Reply:
We need to change 644, and all places where we are sending an error.  

The only places we should have a nonzero user index is on line 632.   And there we should only have a nonzero value if we actually created a user.  And when we create a user, I would expect `createdUserIndex` to be the index we created it at....